src/IB/IBHydrodynamicSurfaceForceEvaluator.cpp

```diff
@@ -291,16 +292,7 @@ IBHydrodynamicSurfaceForceEvaluator::computeHydrodynamicForceTorque(IBTK::Vector
                     n(axis) = signof(phi_upper - phi_lower);
 
                     // Get the relative coordinate from X0
-                    const IBTK::VectorNd side_center = IBTK::IndexUtilities::getSideCenter(*patch, s_i);
-                    IBTK::Vector3d r_vec = IBTK::Vector3d::Zero();
-                    for (unsigned int d = 0; d < NDIM; ++d)
-                    {
-                        r_vec[d] = side_center[d] - X0[d];
-                    }
-
-                    // Get the relative coordinate from X0
-                    r_vec = IBTK::IndexUtilities::getSideCenter<IBTK::Vector3d>(*patch, s_i);
-                    r_vec -= X0;
+                    const auto r_vec = IBTK::IndexUtilities::getSideCenter<IBTK::Vector3d>(*patch, s_i) - X0;
```

Comment:
Would `IBTK::IndexUtilities::getSideCenter<IBTK::Vector3d>(*patch, s_i)` guarantee that r_vec[2] = 0.0 when `NDIM == 2`? This is necessary for the cross product to work properly in 2D.

Reply:
It didn't, but now it does.